ksqldb-functional-tests/src/test/resources/query-validation-tests/row-offset-and-partition.json

```diff
@@ -119,6 +150,622 @@
+        {"topic":  "test_topic", "key": 4, "value": {"foo": "d"}, "timestamp": 8}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"FOO":  "a", "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 4, "value": {"FOO":  "d", "RT": 8}, "timestamp": 8}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, FOO STRING, RT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "Filter and select using ROWPARTITION",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT *, ROWPARTITION AS rp, ROWOFFSET AS ro, ROWTIME AS rt FROM input WHERE ROWPARTITION = 0;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp": 7}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"FOO":  "a", "RP": 0, "RO": 0, "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"FOO":  "b", "RP": 0, "RO": 1, "RT": 6}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"FOO":  "c", "RP": 0, "RO": 2, "RT": 7}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, FOO STRING, RP INT, RO BIGINT, RT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "Filter and select using ROWOFFSET",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT *, ROWTIME AS rt, ROWPARTITION AS rp, ROWOFFSET AS ro FROM input WHERE ROWOFFSET % 3 = 0;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp": 7},
+        {"topic":  "test_topic", "key": 4, "value": {"foo": "d"}, "timestamp": 8}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"FOO":  "a", "RP": 0, "RO": 0, "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 4, "value": {"FOO":  "d", "RP": 0, "RO": 3, "RT": 8}, "timestamp": 8}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, FOO STRING, RT BIGINT, RP INT, RO BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "GREATEST UDF with ROWPARTITION",
```

Comment:
I don't think this and the other two tests (for UDFs) below are necessary, given that we're already testing `CAST()` above.